sdk/communication/AzureCommunicationCalling/Swift_Generated_Interfaces/AzureCommunicationCalling.h

```diff
@@ -0,0 +1,1124 @@
+    open func dealloc()
+
+    
+    /// Video options when starting a call
+    open var videoOptions: VideoOptions!
+
+    
+    /// Audio options when starting a call
+    open var audioOptions: AudioOptions!
+
+    
+    // Class extension begins for StartCallOptions.
+    open var alternateCallerID: PhoneNumberIdentifier
+}
+
+// Class extension ends for StartCallOptions.
+
+
+/// Options when making an outgoing PSTN call
+open class AddPhoneNumberOptions : NSObject {
+
+    public init!()
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    // Class extension begins for AddPhoneNumberOptions.
+    open var alternateCallerID: PhoneNumberIdentifier
+}
+
+// Class extension ends for AddPhoneNumberOptions.
+
+open class AbstractJoinMeetingLocator : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+}
+
+
+/// Options for joining a group call
+open class GroupCallLocator : AbstractJoinMeetingLocator {
+
+    public init!(groupId: UUID!)
+
+    
+    /// The unique identifier for the group conversation
+    open var groupId: UUID!
+}
+
+open class TeamsMeetingCoordinatesLocator : AbstractJoinMeetingLocator {
+
+    public init!(with threadId: String!, organizerId: UUID!, tenantId: UUID!, messageId: String!)
+
+    
+    /// The thread identifier of meeting
+    open var threadId: String! { get }
+
+    
+    /// The organizer identifier of meeting
+    open var organizerId: UUID!
+
+    
+    /// The tenant identifier of meeting
+    open var tenantId: UUID!
+
+    
+    /// The message identifier of meeting
+    open var messageId: String! { get }
+}
+
+open class TeamsMeetingLinkLocator : AbstractJoinMeetingLocator {
+
+    public init!(meetingLink: String!)
+
+    
+    /// The link of the meeting
+    open var meetingLink: String! { get }
+}
+
+open class IncomingCallInformation : NSObject {
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    open var fromDisplayName: String! { get }
+
+    
+    open var hasIncomingVideo: Bool { get }
+
+    
+    // Class extension begins for IncomingCallInformation.
+    open var from: CommunicationIdentifier! { get }
+
+    open var to: CommunicationIdentifier! { get }
+
+    open var callId: UUID { get }
+
+    open class func from(payload: [AnyHashable : Any]!) -> IncomingCallInformation!
+}
+
+// Class extension ends for IncomingCallInformation.
+
+
+/// Options for creating CallAgent
+open class CallAgentOptions : NSObject {
+
+    public init!()
+
+    
+    /// Deallocates the memory occupied by this object.
+    open func dealloc()
+
+    
+    /// Specify the display name of the local participant for all new calls
+    open var displayName: String!
+}
+
+
+/// Call agent created by the CallClient factory method createCallAgent It bears the responsibility of managing calls on behalf of the authenticated user
+open class CallAgent : NSObject {
+
```

Comment:
It seems like this functionality could be merged into the `Call` class

Reply:
Not really `CallClient` is the main entry point which for each token/user can create multiple `CallAgent`s.
`CallAgent` object can now be used to place outgoing calls, join group calls and when there is incoming calls are recieved. They are all `Call` objects which gets created and destroyed as and when there is a call going on.